WineCellar/WineCellar.GUI/DetailedView.xaml.cs

```diff
@@ -0,0 +1,118 @@
+
+namespace WineCellar
+{
+    /// <summary>
+    /// Interaction logic for DetailedView.xaml
+    /// </summary>
+    public partial class DetailedView : Window
+    {
+        private MainWindow mainWindow;
+
+        public int IndexID { get; set; }
+
+        public DetailedView(int value)
+        {
+            InitializeComponent();
+
+            SetData(value);
+        }
+
+        private async void SetData(int index)
+        {
+            List<IWineData> items = await Data.GetAllWines();
+            int lijstIndex = 0;
+
+            foreach (var item in items)
+            {
+                if (index == lijstIndex)
+                {
+                    WineName.DataContext = item.Name;
+                    WineDescription.DataContext = item.Description;
+
+                    WineRating.DataContext = Rating(item.Rating);
+                    WineType.DataContext = item.Type;
+                    WineHarvestYear.DataContext = item.HarvestYear;
+                    WineVolume.DataContext = item.Alcohol + "%";
+                    Taste(item.Taste);
+                    WineCountry.DataContext = item.OriginCountry;
+                    WineLocation.DataContext = item.StorageLocation[0];
+
+                    WineBuy.DataContext = item.BuyPrice;
+                    WineSell.DataContext = item.SellPrice;
+                    WineStock.DataContext = item.Stock;
+
+                    IndexID = item.ID;
+                    break;
+                }
+                else
+                {
+                    lijstIndex++;
+                }
+            }
+        }
+
+        public string Rating(int rating)
+        {
+            StringBuilder sb = new StringBuilder();
+
+            for (int i = 0; i < rating; i++)
+            {
+                sb.Append('★');
+            }
+
+            return sb.ToString();
+        }
+
+        public void Taste(string[] arrayTaste)
+        {
+            foreach (string item in arrayTaste)
+            {
+                ListBoxItem listBoxItem = new ListBoxItem { Content = item };
+                WineTaste.Items.Add(listBoxItem);
+            }
+        }
+
+        private void Button_Click_Terug(object sender, RoutedEventArgs e)
+        {
+            mainWindow = new MainWindow();
+            Application.Current.MainWindow = mainWindow;
+            mainWindow.Show();
+            Close();
+        }
+
+        private async void Button_Click_Verwijderen(object sender, RoutedEventArgs e)
+        {
+            if (MessageBox.Show("Weet u het zeker?", "Verwijderen", MessageBoxButton.YesNo) == MessageBoxResult.Yes)
+            {
+                await Data.DeleteWine(IndexID);
+
+                mainWindow = new MainWindow();
+                Application.Current.MainWindow = mainWindow;
+                mainWindow.Show();
+                Close();
+            }
+        }
+
+        private void Button_Click_Aanpassen(object sender, RoutedEventArgs e)
```

Comment:
Waarom staat hier een lege event?

Reply:
Dat is de knop die leidt naar het 'aanpassen scherm', maar die is er nog niet. Eerst moeten we wachten tot dat het registreren klaar is.